front/Dockerfile

```diff
@@ -1,20 +1,26 @@
 # base image
-FROM node:12.2.0
+FROM node:12.2.0 as build-node
 
-# set working directory
-WORKDIR /app
-
-# add `/app/node_modules/.bin` to $PATH
-ENV PATH /app/node_modules/.bin:$PATH
-
-# install and cache app dependencies
-COPY package.json /app/package.json
+WORKDIR /front
+# COPY ./package*.json /app/
 RUN npm install
 RUN npm install -g typescript
-RUN npm install -g @angular/cli
+RUN npm install -g @angular/cli 
```

Comment:
Jesteś pewien, że po zakomentowaniu kopiowania pliku package.json zainstalujesz zależności?  
`npm install` bez żadnego argumentu instaluje wszystkie zależności z pliku package.json, który jest w tym samym folderze, w którym tę komendę wywołujesz, globalne instalowanie typescripta i angulara jest tutaj bezcelowe.

Reply:
Właśnie dużo problemów mialem z tym frontem i tam bodajże dalej kopiuje wszystko wiec json jest zakomentowany w razie jakbym chciał zoptymalizować wielkość dockera( i nie kopiować na dockera rzeczy ktore nie sa potrzebne). Okej racja, dzięki za pomoc.

Reply:
W wynikowym obrazie dockerowym znajduje się tylko to, co było w ostatnim stage-u, w przypadku waszego frontendu zawartość obrazu nginxa i to, co jest skopiowane z pierwszego stage-a, więc nie ma potrzeby martwić się tym jak dużo plików wrzucicie na etapie budowania, bo te wszystkie tsconfigi, package itd. i tak nie znajdą się na koniec w waszym obrazie.